### ISSUE TITLE
tsc --noEmit is not reporting TS4060: Return type of exported function has or is using private name

### ISSUE THREAD

**Issue by zlk89** [microsoft/TypeScript#39887]
**TypeScript Version:**  3.7.x-dev.201xxxxx

**Search Terms:**
noEmit TS4060
**Code**

```ts
export function test() {
  class TestComponent {
    public render(): string {
      return 'test';
    }
  }
  return TestComponent;
}
```

**Expected behavior:**
Running `yarn tsc --noEmit` should report the error:
`TS4060: Return type of exported function has or is using private name 'TestComponent'.`

**Actual behavior:**
It doesn't show any error

**Playground Link:** 
https://www.typescriptlang.org/play/?noEmitHelpers=true&jsx=2&ts=4.0.0-dev.20200803#

**Related Issues:**
